components/dashboard/src/workspaces/ConnectToSSHModal.tsx

```diff
@@ -55,40 +57,80 @@ interface SSHProps {
-        props.workspaceId,
-        props.workspaceId + ".ssh",
-    )}'`;
+    const [hasSSHKey, setHasSSHKey] = useState(true);
+    const [selectSSHKey, setSelectSSHKey] = useState(true);
+
+    useEffect(() => {
+        getGitpodService()
+            .server.hasSSHPublicKey()
+            .then((d) => {
+                setHasSSHKey(d);
+            })
+            .catch(console.error);
+    }, []);
+
+    const host = props.ideUrl.replace(props.workspaceId, props.workspaceId + ".ssh");
+    const sshAccessTokenCommand = `ssh '${props.workspaceId}#${props.ownerToken}@${host}'`;
+    const sshKeyCommand = `ssh '${props.workspaceId}@${host}'`;
+
     return (
-        <div className="border-t border-b border-gray-200 dark:border-gray-800 mt-2 -mx-6 px-6 py-6">
-            <div className="mt-1 mb-4">
-                <AlertBox>
-                    <p className="text-red-500 whitespace-normal text-base">
+        <>
+            <div className="flex flex-row">
+                <TabMenuItem
+                    key="ssh_key"
+                    name="SSH Key"
+                    selected={selectSSHKey}
+                    onClick={() => {
+                        setSelectSSHKey(true);
+                    }}
+                />
+                <TabMenuItem
+                    key="access_token"
+                    name="Access Token"
+                    selected={!selectSSHKey}
+                    onClick={() => {
+                        setSelectSSHKey(false);
+                    }}
+                />
+            </div>
+            <div className="border-gray-200 dark:border-gray-800 border-b"></div>
+            <div className="space-y-4 mt-4">
+                {!selectSSHKey && (
+                    <Alert type="warning" className="whitespace-normal">
                         <b>Anyone</b> on the internet with this command can access the running workspace. The command
                         includes a generated access token that resets on every workspace restart.
-                    </p>
-                </AlertBox>
-                <InfoBox className="mt-4">
-                    <p className="text-gray-500 whitespace-normal text-base">
-                        Before connecting via SSH, make sure you have an existing SSH private key on your machine. You
-                        can create one using&nbsp;
-                        <a
-                            href="https://en.wikipedia.org/wiki/Ssh-keygen"
-                            target="_blank"
-                            rel="noopener noreferrer"
-                            className="gp-link"
-                        >
-                            ssh-keygen
+                    </Alert>
+                )}
+                {!hasSSHKey && selectSSHKey && (
+                    <Alert type="warning" className="whitespace-normal">
+                        You don't have any public SSH keys in your Gitpod account. You can{" "}
```

Comment:
**praise**: Nice! I came back to this PR to make sure we have this warning here when needed. 👏